### PR TITLE
Enhancement/update dependencies

### PR DESCRIPTION
**Pull request by Enzujp**
Updated Go version to v1.22

### CODE REVIEW

File: .github/workflows/base-ci-goreleaser.yaml [hunk 1]
```diff
@@ -45,7 +45,7 @@ jobs:
       - name: Setup Go
         uses: actions/setup-go@v5
         with:
-          go-version: '~1.21.7'
+          go-version: '~1.22'
```
Comment (TylerHelmuth) [inline .github/workflows/base-ci-goreleaser.yaml:48]: We specify the patch version so should be `~1.22.1`

Comment (Enzujp) [reply .github/workflows/base-ci-goreleaser.yaml]: @TylerHelmuth Fixed it now. Good to go?

File: .gitignore
Comment (TylerHelmuth) [inline .gitignore:1]: Please revert changes to this file

File: .goreleaser.yaml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+release:
```
Comment (TylerHelmuth) [inline .goreleaser.yaml:1]: This change is unrelated

File: .gitignore
Comment (Enzujp) [reply .gitignore:1]: Hello @TylerHelmuth, please confirm that it is as it should be.

File: .goreleaser.yaml (hunk 2 above)
Comment (Enzujp) [reply .goreleaser.yaml:1]: @TylerHelmuth I realize that, It's the .goreleaser file - this branch got updated. Should it be taken out ?

Comment (TylerHelmuth) [reply .goreleaser.yaml:1]: Yes

Comment (Enzujp) [reply .goreleaser.yaml]: Done

**Review by TylerHelmuth — APPROVED** [review]

**Review by mx-psi — APPROVED** [review]

**Review by jpkrohling — COMMENTED** [review]
Shouldn't go.mod be changed as well?

**Comment by Enzujp** [comment]
> Shouldn't go.mod be changed as well?

@jpkrohling I thought about that and then I referenced previous dependency updates #321 and #383 ; go.mod does not need to be manually changed.
